api/graphql/tests/test_applications/test_application_create.py

```diff
@@ -0,0 +1,255 @@
+import datetime
+import json
+
+from assertpy import assert_that
+from django.contrib.auth import get_user_model
+from freezegun import freeze_time
+
+from api.graphql.tests.test_applications.base import ApplicationTestCaseBase
+from applications.models import Application, ApplicationEventStatus, ApplicationStatus
+from applications.tests.factories import CityFactory, OrganisationFactory
+from reservation_units.tests.factories import ReservationUnitFactory
+from reservations.tests.factories import (
+    AbilityGroupFactory,
+    AgeGroupFactory,
+    ReservationPurposeFactory,
+)
+
+
+class ApplicationCreateTestCase(ApplicationTestCaseBase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.age_group = AgeGroupFactory()
+        cls.purpose = ReservationPurposeFactory()
+        cls.ability_group = AbilityGroupFactory()
+        cls.reservation_unit = ReservationUnitFactory()
+        cls.city = CityFactory(name="Hki")
+
+        cls.other_user = get_user_model().objects.create(
+            username="other",
+            first_name="oth",
+            last_name="er",
+            email="oth.er@foo.com",
+        )
+        cls.organisation = OrganisationFactory()
+
+    def get_create_query(self):
+        return """
+            mutation createApplication($input: ApplicationCreateMutationInput!) {
+                createApplication(input: $input){
+                    pk
+                    errors {
+                        messages field
+                    }
+                }
+            }
+        """
+
+    def get_application_data(self, create_events=False):
```

Comment:
Looks like `create_events` parameter is not used 🤔

Reply:
yup, true. Also not `get_event_data` - I started to having the create and update tests in same file so these are leftovers

 - I'll perhaps get rid-off these or make a test which has the events as well in the creation (if we actually support it :sweat_smile: )

Reply:
I added test for adding events in create - I had refactor the code to actually support it. Mut was a miney change in serializer.